### PR TITLE
Add support for recording our UT coverage with coveralls

### PR DESCRIPTION
**Pull request by tomdee**

### CODE REVIEW

**Comment by spikecurtis** [comment]
@tomdee Fails semaphore.  Can you merge master or rebase and push again?

File: Makefile [hunk 1]
```diff
@@ -59,12 +59,15 @@ ut: calicobuild.created
...
+	docker run \
+	-v `pwd`:/code \
 	-v $(CIRCLE_TEST_REPORTS):/circle_output \
+	-e COVERALLS_REPO_TOKEN=$(COVERALLS_REPO_TOKEN) \
 	calico/build bash -c \
 	'/tmp/etcd -data-dir=/tmp/default.etcd/ >/dev/null 2>&1 & \
 	nosetests calico_containers/tests/unit -c nose.cfg \
-	--cover-html-dir=dist --cover-html --with-xunit --xunit-file=/circle_output/output.xml'
+	--with-xunit --xunit-file=/circle_output/output.xml; \
+	[[ ! -z "$$COVERALLS_REPO_TOKEN" ]] && coveralls'
```
Comment (robbrockbank) [inline Makefile]: $$ - doesn't that give PID or something?

**Comment by robbrockbank** [comment]
One question in a comment.

File: Makefile (hunk 1 above)
Comment (tomdee) [inline Makefile]: Nope, it's just escaping the $. It's required in a Makefile.

**Comment by robbrockbank** [comment]
LGTM

**Pull request merged**